dask_match/core.py

```diff
@@ -454,6 +462,61 @@ def _layer(self):
         }
 
 
+class ProjectPartitions(Expr):
+    """Select one or more partitions"""
+
+    _parameters = ["frame", "projection"]
+
+    @property
+    def _meta(self):
+        return self.frame._meta
+
+    @property
+    def partitions(self):
+        if isinstance(self.projection, (list, tuple)):
+            return list(self.projection)
+        raise TypeError
```

Comment:
I think in general we should assume types in Expr code, and ask for user-facing API code to coerce/normalize things.  Thoughts?  

(this comment goes beyond this PR, and is more about general best practices in this repo)